### PR TITLE
[SPARK-38274][BUILD] Upgarde `JUnit4` to `4.13.2` and upgrade corresponding `junit-interface` to `0.13.3`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
The main change of this pr as follows:

- Upgarde `Junit4` to `4.13.2`: https://github.com/junit-team/junit4/blob/HEAD/doc/ReleaseNotes4.13.2.md
- Change  `junit-interface` groupId from `com.novocode` to  `com.github.sbt` due to the organization name has changed to "com.github.sbt" in 0.12 and upgrade `junit-interface` version to `0.13.3` refer to [https://github.com/sbt/junit-interface/#setup](https://github.com/sbt/junit-interface/#setup)

### Why are the changes needed?
Upgarde test framework.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @srowen

**Comment by LuciferYang** [comment]
Copy comparison between `JUnit` and `junit-interface` from [https://github.com/sbt/junit-interface/](https://github.com/sbt/junit-interface/#setup)  as follows:


 junit-interface version  | JUnit version
:-------------------------|:--------------
 0.13.2                   | 4.13.2
 0.13.1                   | 4.13.1
 0.13                     | 4.13
 0.12                     | 4.12

**Review by srowen — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -1144,9 +1144,9 @@
         <scope>test</scope>
       </dependency>
       <dependency>
-        <groupId>com.novocode</groupId>
+        <groupId>com.github.sbt</groupId>
         <artifactId>junit-interface</artifactId>
-        <version>0.11</version>
+        <version>0.13.3</version>
```
Comment (dongjoon-hyun) [inline pom.xml:1149]: It seems that the setup guide is outdated because it still points to `0.13.2`.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: pom.xml (hunk 1 above)
Comment (LuciferYang) [reply pom.xml:1149]: ok ~ I'll try to promote the update of setup

**Comment by LuciferYang** [comment]
thanks all

**Pull request closed without merging**